### PR TITLE
Add check to prevent bad rebin if there are unaligned (2d) bins

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Previous implementation allowed rebinning (e.g., merging) bins of an outer dimension, *keeping* binning of any inner dimensions. However, if those inner bins are unaligned (indicated by presence of a 2d coord) this would give wrong results.

Add checks and test.

### CODE REVIEW

File: dataset/bin.cpp [hunk 1]
```diff
@@ -263,16 +263,24 @@ auto axis_actions(const VariableConstView &var,
...
       axes.emplace_back(AxisAction::Group, dim, groups[groups_dims.index(dim)]);
     else if (edges_dims.contains(dim))
       axes.emplace_back(AxisAction::Bin, dim, edges[edges_dims.index(dim)]);
-    else if (rebin)
+    else if (rebin) {
+      if (array.coords().contains(dim) &&
+          array.coords()[dim].dims().ndim() != 1)
+        throw except::DimensionError(
+            "2-D coordinate " + to_string(array.coords()[dim]) +
+            " conflicting with (re)bin of outer dimension. Try specifying new "
```
Comment (OwenArnold) [inline dataset/bin.cpp:279]: Given that this message will propagate to user. Could you be more explicit? `i.e run bin on dimension {dimension}`

**Review by OwenArnold — CHANGES_REQUESTED** [review]
Fix generally looks good.

File: core/dimensions.cpp [hunk 2]
```diff
@@ -26,7 +26,9 @@ Dimensions::Dimensions(const std::vector<Dim> &labels,
   if (labels.size() != shape.size())
     throw except::DimensionError(
         "Constructing Dimensions: Number of dimensions "
-        "labels does not match shape.");
+        "labels (" +
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply core/dimensions.cpp:29]: Why do you prefer stringstream?

File: dataset/bin.cpp (hunk 1 above)
Comment (SimonHeybrock) [reply dataset/bin.cpp:279]: See update.

File: core/dimensions.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (OwenArnold) [reply core/dimensions.cpp:29]: On further inspection - In this case I don't think it makes much difference as you are either dealing with `const char *` or `std::string` directly. Leave as it is.

**Review by OwenArnold — APPROVED** [review]

**Pull request merged**